src/cargo/ops/cargo_add/mod.rs

```diff
@@ -284,6 +284,7 @@ fn resolve_dependency(
         // Checking for a workspace dependency happens first since a member could be specified
         // in the workspace dependencies table as a dependency
         if let Some(_dep) = find_workspace_dep(dependency.toml_key(), ws.root_manifest()).ok() {
+            check_invalid_ws_keys(dependency.toml_key(), arg)?;
```

Comment:
What is the reason this was added?  This is done in the same commit as everything else but its not clear how its related?

Reply:
I missed adding this to the PR, if this was not here it changes the error message. It prints the error message according to the `toml_key` which changes in the next line (potentially)